app/services/reports/base_report.rb

```diff
@@ -17,6 +17,7 @@ def report_timeout
     end
 
     def transaction_with_timeout
+      Db::EstablishConnection::ReadReplica.call
```

Comment:
Do we need to do something to clean up the read replica connection?

Reply:
Meaning should we throw in an ensure block with something like `ActiveRecord::Base.establish_connection(Rails.env)`?

Reply:
Did you consider a block form of this method? That would make it easier to ensure cleanup, because that method could include the `ensure` block?

Ex:

```
Db::EstablishConnection.in_read_replica do
  ActiveRecord::Base.transaction do
    # ...
  end
end
```